bundler/lib/bundler/yaml_serializer.rb

```diff
@@ -80,6 +84,14 @@ def load(str)
       res
     end
 
+    def strip_comment(val)
+      if val.include?("#") && !val.start_with?("#")
```

Comment:
given this is operating on a new string, we can avoid the array & string allocations by using `val.sub!(/#.*/, "")`